In [1]:
import pandas as pd

path = "../data/raw/accepted_2007_to_2018Q4.csv.gz"


df = pd.read_csv(path, low_memory=False)
print(df.shape)

# The raw CSV has trailing footer/summary rows (e.g. "Total amount funded in
# policy code 1: ...") instead of real loan records, which breaks id's dtype.
df = df[df["id"].apply(lambda x: isinstance(x, int) or str(x).isdigit())].copy()
df["id"] = df["id"].astype("int64")
print(df.shape)

print(df.columns.tolist())
print(df.dtypes)
print(df.head())
print(df.describe())
df.to_parquet("../data/interim/accepted_2007_to_2018Q4.parquet", index=False)

(2260701, 151)
(2260668, 151)
['id', 'member_id', 'loan_amnt', 'funded_amnt', 'funded_amnt_inv', 'term', 'int_rate', 'installment', 'grade', 'sub_grade', 'emp_title', 'emp_length', 'home_ownership', 'annual_inc', 'verification_status', 'issue_d', 'loan_status', 'pymnt_plan', 'url', 'desc', 'purpose', 'title', 'zip_code', 'addr_state', 'dti', 'delinq_2yrs', 'earliest_cr_line', 'fico_range_low', 'fico_range_high', 'inq_last_6mths', 'mths_since_last_delinq', 'mths_since_last_record', 'open_acc', 'pub_rec', 'revol_bal', 'revol_util', 'total_acc', 'initial_list_status', 'out_prncp', 'out_prncp_inv', 'total_pymnt', 'total_pymnt_inv', 'total_rec_prncp', 'total_rec_int', 'total_rec_late_fee', 'recoveries', 'collection_recovery_fee', 'last_pymnt_d', 'last_pymnt_amnt', 'next_pymnt_d', 'last_credit_pull_d', 'last_fico_range_high', 'last_fico_range_low', 'collections_12_mths_ex_med', 'mths_since_last_major_derog', 'policy_code', 'application_type', 'annual_inc_joint', 'dti_joint', 'verification_st

# Understanding Week 0

Here we have a data set that is quite large. The set is ~2.26 million rows with 151 columns.
The raw CSV had a trailing footer/summery rows instead of real loan records. This broke id's dtype. These rows were dropped.

## Data transformation

The file was read compressed (.gz) and saved to interim data as `.parquet`.
A parquet stores data by column allowing reads to be faster.
